src/stream.js

```diff
@@ -21,8 +26,63 @@ module.exports.handler = async (event) => {
 	delete(groupedRecords['deleteMe']); // delete bad events
 
 	for (const records of Object.entries(groupedRecords)) {
-		const latestRecord = records.pop();
+		logger.debug({records}, 'Records');
+		const latestRecord = records.pop().pop();
+		// Tag for a transient record that should be ignored.
+		if (latestRecord.date === '_current') {
+			continue;
+		}
 		logger.debug({latestRecord}, 'Event received');
-	}
 
+		const responses = latestRecord.users && Object.keys(latestRecord.users).map(key => {
+			return {
+				name: `<@${key}>`,
+				answer: latestRecord.users[key] && latestRecord.users[key].answer  === 'fact'
+			};
+		});
+
+		const blockKitParamters = {
+			date: Date.parse(latestRecord.date),
+			questionText: latestRecord.questionText,
+			responses,
+			answer: latestRecord.answer,
+			answerText: latestRecord.answerText
+		};
+
+		const blocks = latestRecord.answer
+			? blockKit.getClosedMessage(blockKitParamters)
+			: blockKit.getOpenMessage(blockKitParamters);
+
+		const postMessageParamters = {
+			date: latestRecord.date,
+			channel: latestRecord.channel,
+			text: '',
+			blocks,
+			ts: latestRecord.slackId
+		};
+
+		const result = latestRecord.slackId
+			? await slack.chat.update(postMessageParamters)
+			: await postMessage(postMessageParamters);
+
+		if (!result.ok) {
+			logger.error({result}, 'Failed to set message');
+		}
+	}
 };
+
+async function postMessage({date, channel, text, blocks}) {
+	const result = await slack.chat.postMessage({
+		channel,
+		text,
+		blocks
+	});
+
+	dynamo.setSlackId({
```

Comment:
I still need to set this for when we `close`. No slackId on `close` command :(

Reply:
I have an idea that may simplify the setting of `slackId` and remove the need to set it in `app.js`.

Currently, you're using the presence or absences of `slackId` to decide if you should do a post or an update. Another approach would be to look at the fields present in a record. If `Object.keys(latestRecord.users)` is empty, you know that no one has voted yet. If `latestRecord.answer` is missing, you know that the question is still open. If both conditions are true, you know `latestRecord` was from the open command so it's the first message and needs to be posted. You can grab the `ts` parameter from the result and set `slackId`.

A problem arises if there is a race between setting `slackId` and someone voting or closing the question. It may happen that you receive a record for voting or closing the question before you receive the record for setting `slackId`. In this case, I would propose simply ignoring any record that does not have `slackId` set (that is not from the open command) as you know a subsequent record will arrive very shortly that does have `slackId` set. When you get the subsequent record you can then do an update of the message.